src/pixelator/annotate/__init__.py

```diff
@@ -229,6 +232,15 @@ def annotate_components(
         json.dump(edgelist_metrics(edgelist), outfile, default=np_encoder)
 
 
+def _cluster_components_using_leiden(adata: AnnData) -> None:
+    """Carry out a leiden clustering on the components."""
+    g = nx.from_scipy_sparse_array(adata.obsp["connectivities"])
+    partitions = leiden(g, resolution=1.0)
+    partitions_df = pd.DataFrame.from_dict(partitions, orient="index").sort_index()
```

Comment:
Just curious about why the need of `sort_index` here?

Reply:
The reason is that the partitions are returned as a dict from `leiden`, and that when creating a data frame from them there is no guarantee of the order they end up in, so I sort them so I can then just assigned them directly as a column to the `adata.obs`.